torch/onnx/symbolic_opset10.py

```diff
@@ -322,39 +322,79 @@ def isfinite(g, input):
+    zero_point = g.op("Cast", zero_point, to_i=sym_help.scalar_type_to_onnx[dtype])
+    scale = g.op("Cast", scale, to_i=torch.onnx.TensorProtoDataType.FLOAT)
+    return sym_help._quantize_helper(g, input, scale, zero_point)
+
+
+def dequantize(g, input):
+    return sym_help._dequantize_helper(g, input)[0]
+
+
 # https://github.com/pytorch/pytorch/wiki/PyTorch-ONNX-exporter#quantized-model-export
 class Quantized:
     domain = "quantized"
 
     # DequantizeLinear was added in opset version 10.
     @staticmethod
-    def linear(g, input_original, weight, weight_scale, weight_zero_point, bias, op_scale, op_zero_point):
-        input_value, input_scale, input_zero_point = sym_help._unpack_tuple(input_original)
+    def linear(g, q_input, q_weight, bias, op_scale, op_zero_point):
         # From https://pytorch.org/docs/master/generated/torch.nn.quantized.functional.linear.html
-        # input (Tensor) – Quantized input of type torch.quint8
-        input_type_dq = torch.onnx.TensorProtoDataType.UINT8
-        input_value = g.op("Cast", input_value, to_i=input_type_dq)
-        input_scale = g.op("Cast", input_scale, to_i=torch.onnx.TensorProtoDataType.FLOAT)
-        input_zero_point = g.op("Cast", input_zero_point, to_i=input_type_dq)
-        input = g.op("DequantizeLinear", input_value, input_scale, input_zero_point)
+        input, input_scale, input_zero_point = sym_help._dequantize_helper(g, q_input)
         # weight (Tensor) – Quantized weight of type torch.qint8
         weight_type_dq = torch.onnx.TensorProtoDataType.INT8
-        weight = g.op("Cast", weight, to_i=weight_type_dq)
-        weight_scale = g.op("Cast", weight_scale, to_i=torch.onnx.TensorProtoDataType.FLOAT)
-        weight_zero_point = g.op("Cast", weight_zero_point, to_i=weight_type_dq)
-        weight = g.op("DequantizeLinear", weight, weight_scale, weight_zero_point)
-        # bias (Tensor) – None or fp32 bias of type torch.float
-        bias = g.op("Cast", bias, to_i=torch.onnx.TensorProtoDataType.FLOAT)
+        weight, weight_scale, weight_zero_point = sym_help._dequantize_helper(g, q_weight, weight_type_dq)
+
         output = linear(g, input, weight, bias)
 
-        if op_scale is None:
-            op_scale = input_scale
-        elif op_scale.type().scalarType() != "Float":
-            op_scale = g.op("Cast", op_scale, to_i=sym_help.cast_pytorch_to_onnx["Float"])
-
-        if op_zero_point is None:
-            op_zero_point = input_zero_point
-        elif op_zero_point.type().scalarType() != "Byte":
-            op_zero_point = g.op("Cast", op_zero_point, to_i=sym_help.cast_pytorch_to_onnx["Byte"])
-        output = g.op("QuantizeLinear", output, op_scale, op_zero_point)
-        return g.op("prim::TupleConstruct", output, op_scale, op_zero_point)
+        return sym_help._quantize_helper(g, output, op_scale, op_zero_point)
+
+    @staticmethod
+    def add(g, x, y, op_scale, op_zero_point):
+        x, x_s, x_zp = sym_help._dequantize_helper(g, x)
```

Comment:
please expand to `x_scale, x_zero_point` for consistency with other variables and other functions in this file.

Same for other lines below

Reply:
I changed to x, _, _ before seeing this comment, since x_scale and x_zero_point are unused.